### PR TITLE
docs: update "Security" docs

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

Updates the documentation of the "referer-restriction", "consumer-restriction", "csrf" and "public-api" Plugins.

Child PR of #6734

### CODE REVIEW

File: docs/en/latest/plugins/public-api.md [hunk 1]
```diff
@@ -23,31 +29,28 @@ title: public-api
...
-| Name | Type | Requirement | Default | Valid | Description |
-| -- | -- | -- | -- | -- | -- |
-| uri | string | optional | "" |   | The uri of the public API. When you set up the route, you can use this to configure the original API uri if it is used in a way that is inconsistent with the original public API uri. |
+The public API added in a custom Plugin is not exposed by default and the user should manually configure a Route and enable the `public-api` Plugin on it.
 
-## Example
+## Attributes
 
-We take the `jwt-auth` token sign API as an example to show how to configure the `public-api` plugin. Also, the `key-auth` will be used to show how to configure the protection plugin for the public API.
+| Name | Type   | Required | Default | Description                                                                                                                                                  |
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/plugins/public-api.md:40]: I thought the default value would be an empty string.

[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/plugins/public-api.md:40]: It was presented in such a way.

**Review by Baoyuantop — COMMENTED** [review]
LGTM

**Comment by navendu-pottekkat** [comment]
Fixed the lint errors.

**Comment by navendu-pottekkat** [comment]
@spacewander @yzeng25 Could you review the PR?

**Review by spacewander — DISMISSED** [review]

**Review by hf400159 — DISMISSED** [review]

**Review by juzhiyuan — APPROVED** [review]

File: docs/en/latest/plugins/referer-restriction.md [hunk 2]
```diff
@@ -23,24 +29,27 @@ title: referer-restriction
 
 ## Description
 
-The `referer-restriction` can restrict access to a Service or a Route by
-whitelisting/blacklisting request header Referrers.
+The `referer-restriction` Plugin can be used to restrict access to a Service or a Route by
+whitelisting/blacklisting the `Referer` request header.
```
Comment (yzeng25) [inline docs/en/latest/plugins/referer-restriction.md:33]: Can you convert it to one line?

Comment (navendu-pottekkat) [reply docs/en/latest/plugins/referer-restriction.md]: Done. Thanks.

File: docs/en/latest/plugins/consumer-restriction.md [hunk 3]
```diff
@@ -23,29 +29,36 @@ title: consumer-restriction
 
 ## Description
 
-The `consumer-restriction` makes corresponding access restrictions based on different objects selected.
+The `consumer-restriction` Plugin allows to set access restrictions based on Consumer, Route, or Service.
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/plugins/consumer-restriction.md]: Done.

**Review by yzeng25 — APPROVED** [review]

**Review by hf400159 — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Pull request merged**